react/AddressForm.tsx

```diff
@@ -1,46 +1,158 @@
+  OptionsField,
+  Display,
+} from './typings/countryRulesTypes.d'
+import rules from './countries/rules'
+import PlaceDetails from './PlaceDetails'
+import { FormattedMessage, defineMessages } from 'react-intl'
+import NumberOption from './components/NumberOption'
 
-interface Props {
-  address: Address
-  summary: LineFragment[][]
+const messages = defineMessages({
+  country: {
+    defaultMessage: '',
+    id: 'place-components.label.country',
+  },
+  state: {
+    defaultMessage: '',
+    id: 'place-components.label.state',
+  },
+  city: {
+    defaultMessage: '',
+    id: 'place-components.label.city',
+  },
+  neighborhood: {
+    defaultMessage: '',
+    id: 'place-components.label.neighborhood',
+  },
+  streetRoadAvenue: {
+    defaultMessage: '',
+    id: 'place-components.label.streetRoadAvenue',
+  },
+  numberOption: {
+    defaultMessage: '',
+    id: 'place-components.label.numberOption',
+  },
+  complement: {
+    defaultMessage: '',
+    id: 'place-components.label.complement',
+  },
+  stateAbbreviation: {
+    defaultMessage: '',
+    id: 'place-components.label.stateAbbreviation',
+  },
+})
+
+type LabelType = keyof (typeof messages)
+
+const getSummaryFields = (summary: LineFragment[][]) => {
+  let summaryFields = new Set()
+  summary.forEach((line: LineFragment[]) => {
+    line.forEach((fragment: LineFragment) => {
+      summaryFields.add(fragment.name)
+    })
+  })
+  return summaryFields
 }
 
-const AddressForm: StorefrontFunctionComponent<Props> = ({
-  address,
-  summary,
-}) => {
-  const { setAddress } = useAddressContext()
+const AddressForm: StorefrontFunctionComponent<{}> = () => {
+  const { address, setAddress } = useAddressContext()
+  const [editing, setEditing] = useState(false)
+  const { fields, display } = rules[address.country]
+  const summary = display.extended as LineFragment[][]
+
+  const [ignoredFields, setIgnoredFields] = useState(
+    getSummaryFields(display.compact)
+  )
+
+  const hasWithoutNumberOption = (label: string) => {
+    return label.length >= 6 && label.substr(label.length - 6) == 'Option'
+  }
+
+  const hasOptions = (field: OptionsField) => {
+    return field.options
+  }
 
   const parseLineFragment = (fragment: LineFragment) => {
-    return address[fragment.name] != null ? (
+    const field = fields[fragment.name as keyof Fields]
+    if (!field) return null
+    const labelName: LabelType = field.label as LabelType
+
+    if (ignoredFields.has(fragment.name)) return null
+    if (address[fragment.name] == null) return null
+    if (hasWithoutNumberOption(labelName)) return <NumberOption showCheckbox />
+
+    const getFieldProps = (fragment: LineFragment) => {
+      const { maxLength, autoComplete, required } = field
+      const label = <FormattedMessage {...messages[labelName]} />
+
+      const onChange = (event: React.ChangeEvent) => {
+        if (
+          event.target instanceof HTMLInputElement ||
+          event.target instanceof HTMLSelectElement
```

Comment:
why do you need to do this check? is there a chance this `onChange` will be called from other elements?